Lock copying issue in prophet test

<!-- Please describe your issue in English. -->

#### Can be reproduced ?
yes

#### Steps:
go vet ./...

#### Expected behavior:
no warning

#### Actual behavior:
components/prophet/prophet_test.go:118:42: literal copies lock value from *c: github.com/matrixorigin/matrixcube/components/prophet/config.Config contains struct{sync.RWMutex; jobProcessors map[github.com/matrixorigin/matrixcube/components/prophet/pb/metapb.JobType]github.com/matrixorigin/matrixcube/components/prophet/config.JobProcessor}

#### Environment:
- Version or commit-id (e.g. v0.1.0 or 8b23a93):
- Hardware parameters:
- OS type:
- Others:

#### Configuration file:


#### Additional context:
- Error message from client:
- Server log:
- Other information:

--- a/components/prophet/config/config.go
+++ b/components/prophet/config/config.go
@@ -59,12 +59,6 @@ type Config struct {
 	ResourceStateChangedHandler     func(res metadata.Resource, from metapb.ResourceState, to metapb.ResourceState) `toml:"-" json:"-"`
 	ContainerHeartbeatDataProcessor ContainerHeartbeatDataProcessor                                                 `toml:"-" json:"-"`
 
-	// Job processor register
-	jobMu struct {
-		sync.RWMutex
-		jobProcessors map[metapb.JobType]JobProcessor
-	} `toml:"-" json:"-"`
-
 	// TODO(fagongzi): the following test-related configurations are moved to a separate struct
 	// Only test can change them.
 	DisableStrictReconfigCheck bool `toml:"-" json:"-"`
@@ -73,6 +67,8 @@ type Config struct {
 	// EnableResponseNotLeader return not leader error for all client request
 	EnableResponseNotLeader bool      `toml:"-" json:"-"`
 	TestCtx                 *sync.Map `toml:"-" json:"-"`
+
+	jobRegister *jobRegister `toml:"-" json:"-"`
 }
 
 // NewConfig creates a new config.

--- a/components/prophet/config/config_job.go
+++ b/components/prophet/config/config_job.go
@@ -1,11 +1,18 @@
 package config
 
 import (
+	"sync"
+
 	"github.com/matrixorigin/matrixcube/components/prophet/core"
 	"github.com/matrixorigin/matrixcube/components/prophet/metadata"
 	"github.com/matrixorigin/matrixcube/components/prophet/pb/metapb"
 	"github.com/matrixorigin/matrixcube/components/prophet/storage"
 )
+
+type jobRegister struct {
+	sync.RWMutex
+	jobProcessors map[metapb.JobType]JobProcessor
+}
 
 // ResourcesAware resources aware
 type ResourcesAware interface {
@@ -31,24 +38,24 @@ type JobProcessor interface {
 
 // RegisterJobProcessor register job processor
 func (c *Config) RegisterJobProcessor(jobType metapb.JobType, processor JobProcessor) {
-	c.jobMu.Lock()
-	defer c.jobMu.Unlock()
+	c.jobRegister.Lock()
+	defer c.jobRegister.Unlock()
 
-	if c.jobMu.jobProcessors == nil {
-		c.jobMu.jobProcessors = make(map[metapb.JobType]JobProcessor)
+	if c.jobRegister.jobProcessors == nil {
+		c.jobRegister.jobProcessors = make(map[metapb.JobType]JobProcessor)
 	}
 
-	c.jobMu.jobProcessors[jobType] = processor
+	c.jobRegister.jobProcessors[jobType] = processor
 }
 
 // GetJobProcessor returns the job handler
 func (c *Config) GetJobProcessor(jobType metapb.JobType) JobProcessor {
-	c.jobMu.RLock()
-	defer c.jobMu.RUnlock()
+	c.jobRegister.RLock()
+	defer c.jobRegister.RUnlock()
 
-	if c.jobMu.jobProcessors == nil {
-		c.jobMu.jobProcessors = make(map[metapb.JobType]JobProcessor)
+	if c.jobRegister.jobProcessors == nil {
+		c.jobRegister.jobProcessors = make(map[metapb.JobType]JobProcessor)
 	}
 
-	return c.jobMu.jobProcessors[jobType]
+	return c.jobRegister.jobProcessors[jobType]
 }

--- a/components/prophet/config/default.go
+++ b/components/prophet/config/default.go
@@ -157,6 +157,10 @@ func (c *Config) Adjust(meta *toml.MetaData, reloading bool) error {
 		return err
 	}
 
+	if c.jobRegister == nil {
+		c.jobRegister = &jobRegister{}
+	}
+
 	return nil
 }
 

